src/core/positioning/nmeagnssreceiver.cpp

```diff
@@ -59,3 +61,27 @@ void NmeaGnssReceiver::stateChanged( const QgsGpsInformation &info )
                                                           info.satPrn, info.satInfoComplete, std::numeric_limits<double>::quiet_NaN(), std::numeric_limits<double>::quiet_NaN(),
                                                           0, QStringLiteral( "nmea" ) );
 }
+
+void NmeaGnssReceiver::nmeaSentenceReceived( const QString &substring )
+{
+  if ( mLogFile.isOpen() )
+  {
+    mLogStream << substring << Qt::endl;
+  }
+}
+
+void NmeaGnssReceiver::handleStartLogging()
+{
+  const QStringList appDataDirs = PlatformUtilities::instance()->appDataDirs();
+  if ( !appDataDirs.isEmpty() )
+  {
+    mLogFile.setFileName( QStringLiteral( "%1/nmea-%2.log" ).arg( appDataDirs.at( 0 ), QDateTime::currentDateTime().toString( QStringLiteral( "yyyy-MM-ddThh:mm:ss" ) ) ) );
+    mLogFile.open( QIODevice::WriteOnly );
```

Comment:
Should we not implement some mechanism for deleting old files? Potentially with the time they could grow to quite some space. Maybe with a setting, or keep the last 10 by default

Reply:
@domi4484 , I was thinking about this too, yet I'm not sure what we should do here. For e.g., say you are sending a field worker to the field for 30 days, and you want to log throughout that period to debug a once-in-a-hundred-use issue, you wouldn't want any logs to be auto deleted.

One alternative would be to not save the setting across QField sessions (i.e. users would _manually_ have to enable logging every time they launch QField). We need to mediate on this :)

Reply:
@nirvn I see your point, I don't know how fast can grow those files, maybe is not at all a problem...

> One alternative would be to not save the setting across QField sessions (i.e. users would _manually_ have to enable logging every time they launch QField). We need to mediate on this :)

I would not do that, otherwise you can be sure that a problem happen only when you forget to enable the option :)

Reply:
let's just leave the simple implementation without deletion. if someone turns this on he/she is aware of it.